src/smoothFUBAR/restrictedFUBAR.jl

```diff
@@ -0,0 +1,142 @@
+struct LogPosteriorRFF{T} # Quite sure this is not needed since we already have it in smoothFUBAR.jl
+    gridcoords::Array{T,2}
+    W::Array{T,2}
+    condlik::Array{T,2}
+    dim::Int
+    unflatten::Function
+    posmask::Vector{T}
+    function LogPosteriorRFF(gridcoords::Array{T,2}, condlik::Array{T,2}, K::Int, σ::T, unflatten, posmask) where T
+        W = randn(T, 2, K ÷ 2) * σ * T(2π)
+        new{T}(gridcoords, W, condlik, K, unflatten, posmask)
+    end
+end
+
+
+function restricted_thetas(ω,ψ, LP::LogPosteriorRFF)
+    WtX = LP.W'LP.gridcoords
+    FF = [cos.(WtX); sin.(WtX)]
+
+    flat_unrestricted_thetas = (ω' * FF)[:]
+
+    matrix_dimension = Integer(sqrt(length(flat_unrestricted_thetas)))
+
+    unrestricted_thetas = reshape(flat_unrestricted_thetas,matrix_dimension, matrix_dimension)
+    
+    interpolated_ψ = softmask(ψ)
+
+    lower_restriction_matrix = (UpperTriangular(reshape(ones(matrix_dimension^2),matrix_dimension,matrix_dimension))) 
+    upper_restriction_matrix = LowerTriangular(reshape(interpolated_ψ .* ones(matrix_dimension^2),matrix_dimension,matrix_dimension)) .- diagm(interpolated_ψ .* ones(matrix_dimension)) # So that we do not OW the diag
+    restriction_matrix = lower_restriction_matrix + upper_restriction_matrix
+
```

Comment:
Use LP.posmask for all of this. It is already flat, matching the theta vec.

Reply:
Done